hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/resolver/FederationNamespaceInfo.java

```diff
@@ -75,4 +76,27 @@ public String getBlockPoolId() {
   public String toString() {
     return this.nameserviceId + "->" + this.blockPoolId + ":" + this.clusterId;
   }
-}
\ No newline at end of file
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj instanceof FederationNamespaceInfo) {
```

Comment:
There is also EqualsBuilder in commons.

Reply:
Thank you for pointing out.
Reimplemented `hashCode` and `equals` with commons: 6124e71